app/controllers/assessments_controller.rb

```diff
@@ -623,6 +623,7 @@ def releaseAllGrades
     num_released = releaseMatchingGrades { |_| true }
 
     if num_released > 0
+      @course.update_course_no_submissions_watchlist_instances
```

Comment:
Also there is an edge case because of the > 0 condition. When nobody submits the assignment, the watchlist will not be added.

Reply:
I was also wondering whether I should put the update statement outside of the >0 condition. I was kind of worried instructors might release twice in a row, by accident, or just to ensure. Do you think the no-one-submits edge case is more likely? I could move the update statement out.

Reply:
Also, since releasing grades for an assessment eventually updates the whole course's watchlist instances, I didn't put the update function call in assessment.rb. This is similar to the grade related condition check where we update the whole course's watchlist instances if an assessment's relevant setting changes.

Reply:
Let's keep the the update statement in the >0 condition then.